integration_tests/sdk/rust/src/main.rs

```diff
@@ -0,0 +1,253 @@
+    };
+    let resp = client.sql_query(rpc_ctx, &query_req).await.unwrap();
+    assert_eq!(resp.affected_rows, 0);
+    let resp_rows = extract_rows_from_sql_query(&resp);
+    let mut expected_rows = Vec::new();
+    if !new_column {
+        let rows = test_data
+            .rows
+            .iter()
+            .take(2)
+            .map(|row| row.iter().take(4).cloned().collect::<Vec<_>>());
+
+        for row in rows {
+            let col_names = test_data.col_names.iter().take(4).cloned();
+            let row = col_names.zip(row.into_iter()).collect::<Vec<_>>();
+            expected_rows.push(row);
+        }
+    } else {
+        let rows = test_data.rows.iter().cloned();
+
+        for row in rows {
+            let col_names = test_data.col_names.iter().cloned();
+            let row = col_names.zip(row.into_iter()).collect::<Vec<_>>();
+            expected_rows.push(row);
+        }
+    }
+
+    let expected = format_rows(&expected_rows);
+    let actual = format_rows(&resp_rows);
+    assert_eq!(expected, actual);
+}
+
+async fn write(
+    client: &Arc<dyn DbClient>,
+    rpc_ctx: &RpcContext,
+    timestamp: i64,
+    test_data: &TestDatas,
+    new_column: bool,
+) {
+    let test_table = format!("test_table_{timestamp}");
+    let mut write_req = WriteRequest::default();
+    let mut points = Vec::new();
+
+    let rows = if !new_column {
+        test_data.rows.iter().take(2).cloned().collect::<Vec<_>>()
+    } else {
+        vec![test_data.rows[2].clone(), test_data.rows[3].clone()]
+    };
+
+    for row in rows {
+        let point = {
+            let timestamp_val = match &row[0] {
+                Value::Timestamp(val) => *val,
+                _ => unreachable!(),
+            };
+            let builder = PointBuilder::new(test_table.clone())
+                .timestamp(timestamp_val)
+                .tag(test_data.col_names[1].clone(), row[1].clone())
+                .field(test_data.col_names[2].clone(), row[2].clone())
+                .field(test_data.col_names[3].clone(), row[3].clone());
+
+            if new_column {
+                builder
+                    .tag(test_data.col_names[4].clone(), row[4].clone())
+                    .field(test_data.col_names[5].clone(), row[5].clone())
+                    .build()
+                    .unwrap()
+            } else {
+                builder.build().unwrap()
+            }
+        };
+        points.push(point);
+    }
+    write_req.add_points(points);
+
+    let resp = client.write(rpc_ctx, &write_req).await.unwrap();
+    assert_eq!(resp.success, 2);
+    assert_eq!(resp.failed, 0);
+}
+
+fn generate_test_datas(timestamp: i64) -> TestDatas {
+    let col_names = vec![
+        "timestamp".to_string(),
+        "old-tag".to_string(),
+        "old-field0".to_string(),
+        "old-field1".to_string(),
+        "new-tag".to_string(),
+        "new-field".to_string(),
+    ];
+
+    let rows = vec![
```

Comment:
It's wired to put all those together. I suggest you can add two fields in TestData
```
{
  OriginRows: Vec<Row>,
  RowsWithAddedColumn: Vec<Row>
}
```

Reply:
I want to generate test data together.
And we just process them for different usage?
Because for `test_add_column` case, we just need to write some new records like:
```
ts, old-tag3, old-field3, new-tag1, new-field1
ts, old-tag4, old-field4, new-tag2, new-field2
```
But when we query, old records with added colmns and new records will be returned and we should check, like:
```
ts, old-tag1, old-field1, default-new-tag, default-new-field
ts, old-tag2, old-field2, default-new-tag, default-new-field
ts, old-tag3, old-field3, new-tag1, new-field1
ts, old-tag4, old-field4, new-tag2, new-field2
```
Even though we split it into two fields, we need to process too.